_schema.graphql

```diff
@@ -414,7 +421,7 @@ enum AnalyticsQueryPeriodEnum {
   SIXTEEN_WEEKS
 }
 
-union AnalyticsRankedEntityUnion = Artwork | Show | Artist
+union AnalyticsRankedEntityUnion = Artwork | PartnerShow | Artist
```

Comment:
The `PartnerShow` type is deprecated, `Show` is the preferred one.

Reply:
@alloy For partner's use case in CMS (e.g. fetching their own non-displayable shows), we need to hit Gravity [partner shows](https://github.com/artsy/gravity/blob/2e667561217b8da7e22e5beab55385dbc3ac6b05/app/api/v1/partners_shows_endpoint.rb#L49) endpoint, instead of [shows](https://github.com/artsy/gravity/blob/2e667561217b8da7e22e5beab55385dbc3ac6b05/app/api/v1/shows_endpoint.rb#L16) endpoint that's for public users. Therefore, we need an authenticated version of the [partnerShowLoader](https://github.com/artsy/metaphysics/blob/ff305c7f5996d264d9465fabbdbb8efe841ac8ce/src/lib/loaders/loaders_without_authentication/gravity.ts#L50), and a type that can resolve with that loader.

The recommended `Show` type is using the [dataloader without authentication](https://github.com/artsy/metaphysics/blob/ff305c7f5996d264d9465fabbdbb8efe841ac8ce/src/schema/v1/show.ts#L668) for pubic use cases, which makes sense. We are hoping to use the other `PartnerShow` for CMS/partner's use cases, but it's also using the [dataloader _without_ authentication](https://github.com/artsy/metaphysics/blob/ff305c7f5996d264d9465fabbdbb8efe841ac8ce/src/schema/v1/partner_show.ts#L372).

It seems like we have to either add a new type (although `PartnerShow` sounds perfect) or make `Show` (or `PartnerShow`) smart enough to conditionally use the new authenticated partner show dataloader (e.g. we might have to differentiate a logged-in user and a logged-in partner). We could miss something here, and would love to get your recommendation in this case!

Reply:
This may be a new pattern once we have more partner's use cases that could be different than logged-in user's use cases.

Reply:
Right, in that case the existing resolver should be made smarter. We shouldn't have 2 types to describe essentially the same modelled data with the only difference being some technical details. The reason is that it makes the graph unintuitive and now associated types need to be added in multiple places.

Reply:
That makes sense! A followup question: Depending on the request (i.e. from unauthenticated user, logged-in user, partner or admin, etc), some fields on the `Show` type may be handled/exposed differently. So we'll likely apply similar technique to resolve them differently?

Reply:
Yup :+1:

You could do it all in the `show` field’s resolver, if there’s not too much differences/overhead, which keeps it close to the different data loaders and thus self-documenting where those differences really are.